torch_geometric/transforms/feature_prop.py

```diff
@@ -0,0 +1,90 @@
+import torch
+from torch_sparse import SparseTensor, matmul
+
+from torch_geometric.data import Data
+from torch_geometric.data.datapipes import functional_transform
+from torch_geometric.nn.conv.gcn_conv import gcn_norm
+from torch_geometric.transforms import BaseTransform
+from torch_geometric.typing import OptTensor
+
+
+@functional_transform('feature_propagation')
+class FeaturePropagation(BaseTransform):
+    r"""The feature propagation operator from the
+    `"On the Unreasonable Effectiveness of Feature
+    propagation in Learning on Graphs with Missing Node Features"
+    <https://arxiv.org/abs/2111.12128>`_ paper
+
+    .. math::
+        \mathbf{X}^{\prime} = \mathbf{D}^{-1/2} \mathbf{A}
+        \mathbf{D}^{-1/2} \mathbf{X},
+
+        \mathbf{X}_{k} = \mathbf{X}^{\prime}_{k}
+
+
+    where missing node features are inferred by known features via propagation.
+
+    .. note::
+
+        For an example of using :class:`~torch_geometric.transforms.
+        FeaturePropagation`, see `examples/feature_propagation.py
+        <https://github.com/pyg-team/pytorch_geometric/blob/master/examples/
+        feature_propagation.py>`_.
+
+    Args:
+        num_iterations (int): The number of propagations. (default: :obj:`40`)
+        missing_mask (Tensor, optional): The mask indicating where
+            node features are missing. (default: :obj:`None`)
+        add_self_loops (bool, optional): If set to :obj:`False`, will not add
+            self-loops to the input graph. (default: :obj:`True`)
+        normalize (bool, optional): Whether to add self-loops and compute
+            symmetric normalization coefficients on the fly.
+            (default: :obj:`True`)
+    """
+    def __init__(self, num_iterations: int = 40,
+                 missing_mask: OptTensor = None, add_self_loops: bool = True,
```

Comment:
Lets make `missing_mask` a required argument.

Reply:
I think the current behaviour with a `missing_mask` is counterintuitive. According to the algorithm, if no features are missing, then everything is left unchanged. On the other hand, if no features are missing (which is the case if `missing_mask` is None), then all features are diffused and change. 

I think it's easier to make `missing_mask` a required argument, so that there is only one behaviour of this algorithm.

Reply:
Yeah I agree, lets drop the option to set it to `None`.

Reply:
Sure,`missing_mask` will be a required argument in the next commit.